packages/perseus/src/widgets/interactive-graphs/reducer/interactive-graph-reducer.ts

```diff
@@ -507,9 +534,181 @@ function boundAndSnapToGrid(
     return snap(snapStep, bound({snapStep, range, point}));
 }
 
-function boundAndSnapToAngle(
+function tooClose(
+    point1: vec.Vector2,
+    point2: vec.Vector2,
+    range: [Interval, Interval],
+) {
+    const safeDistance = 2;
+    const distance = vec.dist(point1, point2);
+
+    return distance < safeDistance;
+}
+
+function boundAndSnapAngleVertex(
+    {
+        range,
+        coords,
+        snapStep,
+    }: {
+        range: [Interval, Interval];
+        coords: [Coord, Coord, Coord];
+        snapStep: vec.Vector2;
+    },
+    {destination}: {destination: vec.Vector2},
+) {
+    // Needed to prevent updating the original coords before the checks for
+    // degenerate triangles and overlapping sides
+    const coordsCopy: [Coord, Coord, Coord] = [...coords];
+
+    const startingVertex = coordsCopy[1];
+    const newVertex = bound({
+        snapStep,
+        range,
+        point: snap(snapStep, destination),
+    });
+    const delta = kvector.add(newVertex, reverseVector(startingVertex));
+
+    let valid = true;
+    const newPoints: Record<string, any> = {};
+
+    _.each([0, 2], function (i) {
+        const oldPoint = coordsCopy[i];
+        let newPoint = kvector.add(oldPoint, delta);
+
+        let angle = GraphUtils.findAngle(newVertex, newPoint);
+        angle *= Math.PI / 180;
+
+        newPoint = constrainToBoundsOnAngle(newPoint, angle, range, snapStep);
```

Comment:
This works but could I just use bound and snap for the vertex since it's using the snap step